src/rgw/rgw_rados.cc

```diff
@@ -1329,6 +1349,12 @@ int RGWRados::initialize()
     cct->_conf.get_val<double>("rgw_inject_notify_timeout_probability");
   max_notify_retries = cct->_conf.get_val<uint64_t>("rgw_max_notify_retries");
 
+  auto ec = init_neo();
+  if (ec) {
+    ldout(cct, 0) << "ERROR: failed to init RADOS (ec=" << ec << ")" << dendl;
+    ceph::from_error_code(ec);
+  }
```

Comment:
this error needs to be returned as a failure. consider adding `[[nodiscard]]` to `ceph::from_error_code()` so the compiler can help catch bugs like this!

Reply:
Fixed, thank you.